api/data_explorer/controllers/facets_controller.py

```diff
@@ -51,9 +50,20 @@ def deserialize(filter_arr):
     for facet_filter in filter_arr:
         filter_str = urllib.unquote(facet_filter).decode('utf8')
         key_val = filter_str.split('=')
+        facet_name = key_val[0]
+        facet_value = key_val[1]
+        es_facet = current_app.config['ELASTICSEARCH_FACETS'][facet_name]
+        if isinstance(es_facet, HistogramFacet):
+            facet_value = bucket_string_to_number(facet_value)
         if len(key_val) == 2:
```

Comment:
Isn't len(key_val) always 2?

Reply:
Yes, but since the 'name=value' syntax is just contractual between the server implementation and the client, there's nothing stopping anything from sending an incorrectly formatted filter string, so this is just a sanity check to make sure we only act if we're getting what we expected.

Reply:
Don't need to make any changes, just a comment in general -- it's a little weird to me to see "if len(key_val) == 2:" but not do anything if the condition isn't true. If you're going to make the check, throw an error if the check fails. (Or don't make the check.)